nomad/worker.go

```diff
@@ -874,12 +876,17 @@ func (w *Worker) shouldResubmit(err error) bool {
 // backoff if the server or the worker is shutdown.
 func (w *Worker) backoffErr(base, limit time.Duration) bool {
 	w.setWorkloadStatus(WorkloadBackoff)
-	backoff := (1 << (2 * w.failures)) * base
-	if backoff > limit {
-		backoff = limit
-	} else {
-		w.failures++
+
+	backoff := w.failureBackoff
+	if w.failureBackoff < limit {
```

Comment:
isn't this always true, as long as `limit` is greater than 0?  since nothing is setting `w.failureBackoff` to something different from its default zero value.

Reply:
D'oh, I didn't set the `w.failureBackoff` below. That's just a bug. Will fix.